PNA.mdk

```diff
@@ -594,6 +594,42 @@ Calling `send_to_port(x)` modifies hidden state for this packet, so
 that the packet will be sent to the vport with id `x` in the host,
 without being looped back.
 
+
+### Extern function `recirculate`
+
+The extern function `recirculate` is used to cause a packet, after it
+is deparsed, to recirculate, &ie; to go back and become an input to
+the main parser again.
+
+~ Begin P4Example
+[INCLUDE=pna.p4:recirculate]
+~ End P4Example
+
+See Section [#sec-extern-restrctions] for what is meant by
+`recirculate` being restricted to being called only from the main
+control.
+
+Note that it is still possible to preserve more than 32 bits of
+metadata with a packet after it has been recirculated.  For example,
+one way is to explicitly use the `setValid` operation on one or more
+headers, `emit` them in the deparser, and parser them in your main
+parser.
+
+~Begin TBD
+We should decide and document the PNA behavior when arbitrary
+combinations of the functions `drop_packet`, `send_to_port`,
+`recirculate`, and `mirror_packet` are called in one execution of the
+main control, and also if the _order_ of such calls affects the
+behavior, e.g. if the result of calling `drop_packet` then
+`recirculate` is the same as calling `recirculate` then `drop_packet`,
+or different.  Alternately, we can explicitly say "every PNA target
+should define and document their behavior on arbitrary combinations of
+these operations, if they wish".  P4 developers would not find that as
```

Comment:
AI Andy: Add a note about one possibility: last-one-wins among drop_packet send_to_port recirculate, but mirror_packet always takes effect on top of the last of those 3 ops.